cmd/server-main.go

```diff
@@ -118,26 +118,26 @@ func serverHandleCmdArgs(ctx *cli.Context) {
 
 	// Server address.
 	serverAddr := ctx.String("address")
-	logger.FatalIf(CheckLocalServerAddr(serverAddr), "Invalid address ‘%s’ in command line argument.", serverAddr)
+	logger.FatalIf(CheckLocalServerAddr(serverAddr), "Unable to validate passed arguments")
 
 	var setupType SetupType
 	var err error
 
 	if len(ctx.Args()) > serverCommandLineArgsMax {
-		logger.FatalIf(errInvalidArgument, "Invalid total number of arguments (%d) passed, supported upto 32 unique arguments", len(ctx.Args()))
+		uErr := uiErrInvalidErasureEndpoints(nil).Msg(fmt.Sprintf("Invalid total number of endpoints (%d) passed, supported upto 32 unique arguments", len(ctx.Args())))
+		logger.FatalIf(uErr, "Unable to validate passed endpoints")
 	}
 
 	globalMinioAddr, globalEndpoints, setupType, globalXLSetCount, globalXLSetDriveCount, err = createServerEndpoints(serverAddr, ctx.Args()...)
-	logger.FatalIf(err, "Invalid command line arguments server=‘%s’, args=%s", serverAddr, ctx.Args())
+	logger.FatalIf(err, "Unable to validate passed endpoints")
 
 	globalMinioHost, globalMinioPort = mustSplitHostPort(globalMinioAddr)
-	if runtime.GOOS == "darwin" {
-		// On macOS, if a process already listens on LOCALIPADDR:PORT, net.Listen() falls back
-		// to IPv6 address ie minio will start listening on IPv6 address whereas another
-		// (non-)minio process is listening on IPv4 of given port.
-		// To avoid this error sutiation we check for port availability only for macOS.
-		logger.FatalIf(checkPortAvailability(globalMinioPort), "Port %d already in use", globalMinioPort)
-	}
+
+	// On macOS, if a process already listens on LOCALIPADDR:PORT, net.Listen() falls back
+	// to IPv6 address ie minio will start listening on IPv6 address whereas another
+	// (non-)minio process is listening on IPv4 of given port.
+	// To avoid this error sutiation we check for port availability.
```

Comment:
This change is not correct, we should only check for port availability specifically on darwin, on Linux SO_REUSEADDR is not allowed by default.

Reply:
More than `"Unable to start the server"`  this message, previous message is still correct and valid.

Reply:
> This change is not correct, we should only check for port availability specifically on darwin, on Linux SO_REUSEADDR is not allowed by default.

Can you elaborate more ? I tested this style and it seems to be working correctly. I think golang is already passing SO_REUSEADDR by default (https://golang.org/src/net/sockopt_linux.go, line 25)

Reply:
> More than "Unable to start the server" this message, previous message is still correct and valid.

With the new error log, the port reserved issue will still be shown (it is not removed actually), it will just come from 
 lower level.

Reply:
Oh actually not SO_REUSEADDR Darwin has a peculiar behavior where an app can listen on IPv4 and another app can listen on IPv6 .. so we don't need to check port availability here since Linux verifies for both.